host/xtest/pkcs11_1000.c

```diff
@@ -8016,3 +8016,173 @@ static void xtest_pkcs11_test_1025(ADBG_Case_t *c)
+		128, &oaep_params,
+	};
+	CK_MECHANISM rsa_aes_key_wrap_mechanism = {
+		CKM_RSA_AES_KEY_WRAP, &rsa_aes_params, sizeof(rsa_aes_params),
+	};
+
+	uint8_t g_val[RSA_AES_SAMPLE_KEY_SIZE] = { 0 };
+	CK_ULONG key_len = 0;
+	CK_ATTRIBUTE get_template[] = {
+		{ CKA_VALUE_LEN, &key_len, sizeof(key_len) },
+		{ CKA_VALUE, g_val, sizeof(g_val) },
+	};
+	uint8_t g_unwrapped_val[RSA_AES_SAMPLE_KEY_SIZE] = { 0 };
+	CK_ULONG unwrapped_key_len = 0;
+	CK_ATTRIBUTE get_template_unwrapped[] = {
+		{ CKA_VALUE_LEN, &unwrapped_key_len,
+		  sizeof(unwrapped_key_len) },
+		{ CKA_VALUE, g_unwrapped_val, sizeof(g_unwrapped_val) },
+	};
+
+	uint8_t buf[RSA_AES_WRAPPED_KEY_SIZE] = { 0 };
+	CK_ULONG size = 0;
+
+	rv = init_lib_and_find_token_slot(&slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	rv = init_test_token(slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = init_user_test_token(slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_lib;
+
+	rv = C_Login(session, CKU_USER,	test_token_user_pin,
+		     sizeof(test_token_user_pin));
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_GenerateKeyPair(session, &keygen_mechanism,
+			       wrapping_key_template,
+			       ARRAY_SIZE(wrapping_key_template),
+			       unwrapping_key_template,
+			       ARRAY_SIZE(unwrapping_key_template),
+			       &wrapping_key, &unwrapping_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_session;
+
+	rv = C_GenerateKey(session, &cktest_aes_keygen_mechanism,
+			   key_template, ARRAY_SIZE(key_template),
+			   &key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto close_session;
+
+	Do_ADBG_BeginSubCase(c, "Test key wrap with RSA AES");
+
+	size = sizeof(buf);
+	rv = C_WrapKey(session, &rsa_aes_key_wrap_mechanism, wrapping_key, key,
+		       buf, &size);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_GetAttributeValue(session, key, get_template,
+				 ARRAY_SIZE(get_template));
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	Do_ADBG_BeginSubCase(c, "Test key unwrap with RSA AES");
+
+	rv = C_UnwrapKey(session, &rsa_aes_key_wrap_mechanism, unwrapping_key,
+			 buf, size, new_key_template,
+			 ARRAY_SIZE(new_key_template), &unwrapped_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_GetAttributeValue(session, unwrapped_key, get_template_unwrapped,
+				 ARRAY_SIZE(get_template_unwrapped));
+	if (!ADBG_EXPECT_CK_OK(c, rv) ||
+	    !ADBG_EXPECT_BUFFER(c, g_unwrapped_val, unwrapped_key_len,
+				g_val, key_len))
+		goto out;
+
+	rv = C_DestroyObject(session, unwrapped_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+out:
+	Do_ADBG_EndSubCase(c, NULL);
+close_session:
+	ADBG_EXPECT_CK_OK(c, C_CloseSession(session));
+
+close_lib:
+	ADBG_EXPECT_CK_OK(c, close_lib());
+}
+ADBG_CASE_DEFINE(pkcs11, 1026, xtest_pkcs11_test_1026,
+		 "PKCS11: RSA AES Key Wrap/UnWrap tests");
```

Comment:
Capitalization is a bit funny here, how about "PKCS11: RSA AES key wrap/unwrap tests"?

Reply:
name of test case #1020 has this capitalization, so I did the same way. But it's corrected now.

I've extended tests a bit, so I must ask you to take a look once again please.